app.js

```diff
@@ -2,48 +2,98 @@ const express = require("express");
 const logger = require("morgan");
 
 const app = express();
+const Recipe = require("./models/Recipe.model");
 
 // MIDDLEWARE
 app.use(logger("dev"));
 app.use(express.static("public"));
 app.use(express.json());
 
+const mongoose = require("mongoose");
+const { restart } = require("nodemon");
 
 // Iteration 1 - Connect to MongoDB
 // DATABASE CONNECTION
+const MONGODB_URI = "mongodb://127.0.0.1:27017/express-mongoose-recipes-dev";
 
-
+mongoose
+  .connect(MONGODB_URI)
+  .then((x) =>
+    console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`)
+  )
+  .catch((err) => console.error("Error connecting to mongo", err));
 
 // ROUTES
-//  GET  / route
-app.get('/', (req, res) => {
-    res.send("<h1>Lab | Express Mongoose Recipes</h1>");
+//  GET  / route - This is just an example route
+app.get("/", (req, res) => {
+  res.send("<h1>LAB | Express Mongoose Recipes</h1>");
 });
 
-
-//  Iteration 4 - Create recipe route
+//  Iteration 3 - Create a Recipe route
 //  POST  /recipes route
+app.post("/recipes", (req, res) => {
+  Recipe.create(req.body)
+    .then((createdRecipe) => {
+      res.status(201).json(createdRecipe);
+    })
+    .catch(() => {
+      res.status(500).json({ message: "Error" });
+    });
+});
 
-
-//  Iteration 5 - Read all recipes
+//  Iteration 4 - Get All Recipes
 //  GET  /recipes route
+app.get("/recipes", (req, res) => {
+  Recipe.find()
+    .then((allRecipes) => {
+      res.status(200).json(allRecipes);
+    })
+    .catch(() => {
+      res.status(500).json({ message: "Error" });
+    });
+});
 
-
-//  Iteration 6 - Read a single recipe
+//  Iteration 5 - Get a Single Recipe
 //  GET  /recipes/:id route
 
+app.get("/recipes/:id", (req, res) => {
+  Recipe.findById(req.params.id)
+    .then((oneRecipe) => {
+      res.status(200).json(oneRecipe);
+    })
+    .catch(() => {
+      res.status(500).json({ message: "Error" });
+    });
+});
 
-//  Iteration 7 - Update a single recipe
+//  Iteration 6 - Update a Single Recipe
 //  PUT  /recipes/:id route
 
+app.put("/recipes/:id", (req, res) => {
+  Recipe.findByIdAndUpdate(req.params.id, req.body, { new: true })
```

Comment:
Same case here :shipit: for the req.body, also something that you can do is for the id in the params 
```suggestion
app.put("/recipes/:id", (req, res) => {
const { id } = req.params
  Recipe.findByIdAndUpdate( id, req.body, { new: true })
```